core/tests/unit/ldml/test_context_normalization.cpp

```diff
@@ -109,12 +109,12 @@ void test_context_normalization_hefty() {
 }
 
 void test_context_normalization_invalid_unicode() {
-                                          // unpaired surrogate     illegal
-  km_core_cp const application_context[] = { 0xDC01, 0x0020, 0x0020, 0xFFFF, 0x0000 };
-  km_core_cp const cached_context[] =      { 0xDC01, 0x0020, 0x0020, 0xFFFF, 0x0000 };
+                                      // unpaired surrogate             illegal
+  km_core_cp const application_context[] = {    0xDC01, 0x0020, 0x0020, 0xFFFF, 0x0000 };
```

Comment:
No change to test case

Reply:
You changed cached_context in this test case and also compared application_context+1 on line 117. The point of this test was to verify behaviour of illegal unicode in the context. As we have not yet implemented #10392 for mid-context unpaired surrogates, we should not be losing the test of this behaviour, hence asking for a separate test for the specific leading trailing surrogate char.

Reply:
I'll split it.

Reply:
I won't expect line 114 or 117 to ever pass as-is with this change. But let's leave that as a separate concern.   I also took the 0xFFFF out of the updated, separate test case.

Reply:
> I won't expect line 114 or 117 to ever pass as-is with this change. But let's leave that as a separate concern. I also took the 0xFFFF out of the updated, separate test case.

Yes, I think Marcs suggestion was a) for a separate test case but b) update the current one by inserting some valid chars at the context so that the invalid char is now in the middle some where. Therefore 114 and 117 will be ok and the intent of the test is preserved.  
Yes ok to merge I will just add a note for #10392